Sources/PackageCollectionsSigning/Key/Key+EC.swift

```diff
@@ -0,0 +1,61 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+
+import Crypto
+
+typealias CryptoECPrivateKey = P256.Signing.PrivateKey
+typealias CryptoECPublicKey = P256.Signing.PublicKey
+
+struct ECPrivateKey: PrivateKey {
+    let underlying: CryptoECPrivateKey
+
+    init<Data>(pem data: Data) throws where Data: DataProtocol {
+        let pem = String(decoding: data, as: UTF8.self)
+        #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+        if #available(macOS 11, *) {
+            self.underlying = try CryptoECPrivateKey(pemRepresentation: pem)
+        } else {
+            let data = try KeyUtilities.stripHeaderAndFooter(pem: pem)
+            // From the output of `openssl pkey -in eckey.pem -text -noout`, the P256 private key is 32-byte long.
+            // PEM format is 7-byte pre_string || 32-byte private key || 14-byte mid_string || 65-byte public key
+            // See: https://stackoverflow.com/questions/48101258/how-to-convert-an-ecdsa-key-to-pem-format
+            self.underlying = try CryptoECPrivateKey(rawRepresentation: data.dropFirst(7).prefix(32))
```

Comment:
This seems unsafe, it's missing a lot of validation about what kind of key this all is.

Do we have to support macOS Catalina?

Reply:
> Do we have to support macOS Catalina?

Yes we do

I rewrote this code. See https://github.com/apple/swift-package-manager/pull/3238#discussion_r571873897